src/hooks/useProperties.ts

```diff
@@ -0,0 +1,30 @@
+import { JSONSchema4 } from 'json-schema';
+import { useMemo } from 'react';
+import { SchemaTreeNode } from '../types';
+import { getProperties, IFilterOptions } from '../utils/getProperties';
+
+export const useProperties = (
+  schema: JSONSchema4,
+  dereferencedSchema: JSONSchema4 | undefined,
+  options: IFilterOptions
+) => {
+  return useMemo(
+    () => {
+      let { limitPropertyCount } = options;
+      const nodes: SchemaTreeNode[] = [];
+      let isOverflow = false;
+
+      for (const node of getProperties(schema, dereferencedSchema, options)) {
+        if (limitPropertyCount !== undefined && limitPropertyCount-- < 0) {
```

Comment:
Mutating `limitPropertyCount` in the `if` statement smells a bit. Suggest putting on it's own line.